src/components/Carousel.js

```diff
@@ -653,14 +653,17 @@ export default class Carousel extends Component {
   }
 
   render() {
+    const { value, children, slides } = this.props;
```

Comment:
`value` prop should be retrieved using `this. getCurrentValue` method (now it just returns `this.props.value` but the behaviour may change in the future).
There is a method to retrieve a list of slides, which covers all cases (including e.g. one child): `this.getChildren`. It should be used instead of manually checking slides and children.